zipkin-cassandra/src/test/scala/com/twitter/zipkin/cassandra/CassandraSpanStoreFactorySpec.scala

```diff
@@ -46,4 +46,38 @@ class CassandraSpanStoreFactorySpec extends FunSuite with Matchers {
       asList(new InetSocketAddress("1.1.1.1", 9143), new InetSocketAddress("2.2.2.2", 9042))
     )
   }
+
+  test("creatingClusterBuilder with authentication") {
+    TestFactory.nonExitingMain(Array(
+      "-zipkin.store.cassandra.auth", "true",
+      "-zipkin.store.cassandra.user", "user",
+      "-zipkin.store.cassandra.password", "pass"
+    ))
+
+    val builder = TestFactory.createClusterBuilder()
+    val authProvider     = get_privates(builder, "authProvider")
+    val user = get_privates(authProvider, "username")
+    val pass = get_privates(authProvider, "password")
+    user should be("user")
+    pass should be("pass")
+  }
+
+  test("creatingClusterBuilder without authentication") {
+    TestFactory.nonExitingMain(Array(
+      "-zipkin.store.cassandra.auth", "false"
+    ))
+
+    val builder = TestFactory.createClusterBuilder()
+    val authProvider     = get_privates(builder, "authProvider")
+    val user = get_privates(authProvider, "username")
```

Comment:
I dug through the code a bit.. 

should be able to do this instead. `TestFactory.createClusterBuilder().getConfiguration(). getAuthProvider()`

cast as `PlainTextAuthenticator` then `getCredentials()` should match.

That's more refactor save and doesn't use reflection. If doesn't work, lemme know!